### PR TITLE
feat: Add Gong Read connector

### LINKED ISSUE #247: Add Gong Read capability to existing connector

### PR DESCRIPTION
**Pull request by Listener430**
+ Linter
+ Tests passed

Test description: Gong Deep (Read) Connector. Gets the list of call recordings  (calls), the required field is 'url'. 
The total number of call records (calls) available in the sandbox is 179. Gong always returns the first 100 records and a cursor in case there's more. 

Test location: test\gong\read\read.go

gong_read-top_of_page1

The bottom of the page contains the link to the next page 
gong_read-bottom_page1

Unit tests results
gong_unit_tests

closes #247

### CODE REVIEW

File: test/gong/read/read.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	if err := godotenv.Load(); err != nil {
+		slog.Warn("No .env file provided")
+	}
+
+	gong := GetGongConnector(context.Background(), DefaultCredsFile)
+
+	config := connectors.ReadParams{
+		ObjectName: "calls", // could be calls, users
+		Fields:     []string{"url"},
+		//NextPage:   "eyJhbGciOiJIUzI1NiJ9.eyJjYWxsSWQiOjQ5NTM3MDc2MDE3NzYyMzgzNjAsInRvdGFsIjoxNzksInBhZ2VOdW1iZXIiOjAsInBhZ2VTaXplIjoxMDAsInRpbWUiOiIyMDIyLTA5LTEzVDA5OjMwOjAwWiIsImV4cCI6MTcxNDQ3NjU4MX0.slRd0i1iPnBHnOxgVPhCREIJmmxkUBJOd6f0N9lHfHs",
```
Comment (laurenzlong) [inline test/gong/read/read.go:95]: Can you remove this line?

File: gong/connector.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	conn = &Connector{
+		Client:    params.client,
+		BaseURL:   providerInfo.BaseURL,
+		APIModule: params.APIModule,
+	}
+
+	conn.setBaseURL(providerInfo.BaseURL)
+	conn.Client.HTTPClient.ErrorHandler = interpreter.ErrorHandler{
```
Comment (laurenzlong) [inline gong/connector.go:65]: We can't expect our customers to be modifying this code to be adding error handling logic. Can you revise this to:

```
	conn.Client.HTTPClient.ErrorHandler = conn.interpretError
```

File: gong/model.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package gong
+
+type User struct {
```
Comment (laurenzlong) [inline gong/model.go:3]: I don't see these structs used anywhere

File: gong/test/read.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (laurenzlong) [inline gong/test/read.json:1]: What's this file for? I don't think I see it being used.

**Review by laurenzlong — COMMENTED** [review]
@Listener430 can you please revise this PR to include screenshots of test output in the PR description?

File: gong/errors.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"errors"
+)
+
+var (
+	ErrMissingClient = errors.New("JSON http client not set")
+	ErrNotArray      = errors.New("results data is not an array")
+	ErrNotObject     = errors.New("record is not an object")
+	ErrNotString     = errors.New("next is not a string")
+	ErrGeneral       = errors.New("general error")
```
Comment (laurenzlong) [inline gong/errors.go:12]: `ErrGeneral` and `ErrNotString` doesn't seem used anywhere. (Also, we want to avoid unhelpful errors like `ErrGeneral`)

**Review by laurenzlong — CHANGES_REQUESTED** [review]

File: gong/model.go (hunk 3 above)
Comment (Listener430) [reply gong/model.go:3]: fixed

File: gong/errors.go (hunk 5 above)
Comment (Listener430) [reply gong/errors.go:12]: fixed

File: gong/connector.go (hunk 2 above)
Comment (Listener430) [reply gong/connector.go:65]: fixed

**Review by laurenzlong — CHANGES_REQUESTED** [review]
@Listener430 sorry for the delayed re-review, it seems like there's now a lot of unrelated commits in this PR, and the changed files included a lot of other changes also. Can you please fix the Git history? It may be easier to make a new PR altogether with just the Gong related files. And please re-request review from me then.

**Comment by RajatPawar** [comment]
@Listener430 bump

**Comment by Listener430** [comment]
Closing as a duplicte of https://github.com/amp-labs/connectors/pull/431

**Pull request closed without merging**